src/components/image-card-grid/image-card-grid.vue

```diff
@@ -29,20 +34,25 @@
   </section>
 </template>
 
-<script>
-  export default {
-    props: {
-      title: {
-        type: String,
-        required: true,
-      },
-      items: {
-        type: Array,
-        required: false,
-        default: () => [],
-      },
-    },
-  }
+<script setup lang="ts">
+import { BackgroundColor } from '../../types/index.d'
+
+type Props = {
+  title: string
+  items: {
+    id: string
+    title: string
+    body: object
+    image: {
+      url: string
+    }
+  }[]
+  backgroundColor: BackgroundColor
+}
+
+withDefaults(defineProps<Props>(), {
+  backgroundColor: BackgroundColor.None
+})
```

Comment:
Since Vue 3.3 we can declare defaults with destructuring I think, something like:
```suggestion
const props = defineProps({
  title: string
  items: {
    id: string
    title: string
    body: object
    image: {
      url: string
    }
  }[]
  backgroundColor: BackgroundColor = BackgroundColor.None,
})
```

Not sure if it's possible with types though... Ignore this comment if it isn't.

Reply:
I'm getting a whole load of typescript errors:

```
No overload matches this call.
  Overload 1 of 3, '(props: string[]): Readonly<{ [x: string]: any; }>', gave the following error.
  Overload 2 of 3, '(props: ComponentObjectPropsOptions<Data>): Readonly<ExtractPropTypes<ComponentObjectPropsOptions<Data>>>', gave the following error.ts(2769)
runtime-core.d.ts(436, 61): The expected type comes from property 'backgroundColor' which is declared here on type 'ComponentObjectPropsOptions<Data>'
```